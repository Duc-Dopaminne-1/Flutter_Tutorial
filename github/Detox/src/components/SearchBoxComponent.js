import PropTypes from 'prop-types';
import React from 'react';
import {Platform, StyleSheet, Text, View, ViewPropTypes} from 'react-native';

import {CONSTANTS, MAX_LENGTH} from '../assets/constants';
import {SIZES} from '../assets/constants/sizes';
import {ICONS} from '../assets/icons';
import CustomIcon from '../assets/icons/CustomIcon';
import {COLORS} from '../assets/theme/colors';
import {FONTS} from '../assets/theme/fonts';
import {small, tiny} from '../assets/theme/metric';
import DebounceInput from './DebounceInput';

const containerPadding = Platform.OS === 'ios' ? 12 : tiny;
const styles = StyleSheet.create({
  viewContainer: {
    flexDirection: 'row',
    paddingHorizontal: small,
    paddingVertical: containerPadding,
    borderRadius: 4,
    borderWidth: SIZES.BORDER_WIDTH_1,
    borderColor: COLORS.PRIMARY_A100,
    alignItems: 'center',
  },
  textSearch: {
    ...FONTS.regular,
    flex: 1,
    color: COLORS.BLACK_33,
    fontSize: 16,
    paddingLeft: 4,
  },
});

const SearchBoxComponent = ({
  style,
  value,
  onChangeKeyword,
  textSearchStyle,
  placeholder,
  delayTimeout,
  iconColor = COLORS.PRIMARY_A100,
}) => {
  return (
    <View style={[styles.viewContainer, style]}>
      <DebounceInput
        value={value}
        onChangeText={onChangeKeyword}
        style={[styles.textSearch, textSearchStyle]}
        maxLength={MAX_LENGTH.default}
        placeholder={placeholder}
        autoCorrect={false}
        autoCapitalize="none"
        delayTimeout={delayTimeout}
      />
      <CustomIcon name={ICONS.SEARCH} size={20} color={iconColor} />
    </View>
  );
};

SearchBoxComponent.propTypes = {
  style: ViewPropTypes.style,
  value: PropTypes.string,
  onChangeKeyword: PropTypes.func,
  textSearchStyle: Text.propTypes.style,
  placeholder: PropTypes.string,
  delayTimeout: PropTypes.number,
};

SearchBoxComponent.defaultProps = {
  value: '',
  onChangeKeyword: () => {},
  placeholder: '',
  delayTimeout: CONSTANTS.DEFAULT_KEYWORD_TIME_OUT,
};

export default SearchBoxComponent;
