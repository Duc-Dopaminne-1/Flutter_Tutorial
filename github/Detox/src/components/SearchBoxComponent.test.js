import React from 'react';
import renderer from 'react-test-renderer';

import SearchBoxComponent from './SearchBoxComponent';

it('renders correctly', () => {
  const tree = renderer.create(<SearchBoxComponent />).toJSON();
  expect(tree).toMatchSnapshot();
});
