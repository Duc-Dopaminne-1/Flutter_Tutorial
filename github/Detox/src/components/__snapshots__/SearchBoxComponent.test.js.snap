// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderColor": "#26894F",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "paddingHorizontal": 8,
        "paddingVertical": 12,
      },
      undefined,
    ]
  }
>
  <TextInput
    autoCapitalize="none"
    autoCorrect={false}
    delayTimeout={500}
    maxLength={128}
    onChangeText={[Function]}
    placeholder=""
    style={
      Array [
        Object {
          "color": "#333333",
          "flex": 1,
          "fontFamily": "Nunito-Regular",
          "fontSize": 16,
          "includeFontPadding": true,
          "paddingLeft": 4,
        },
        undefined,
      ]
    }
    value=""
  />
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#26894F",
          "fontSize": 20,
        },
        undefined,
        Object {
          "fontFamily": "icomoon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    Ó§ç
  </Text>
</View>
`;
